src/main/scala/org/apache/spark/sql/catalyst/plans/logical/merge.scala

```diff
@@ -266,7 +266,7 @@ object MergeInto {
             // plan.
             fakeTargetPlan.output.map(_.name).map { tgtColName =>
               val resolvedExpr =
-                resolveOrFail(UnresolvedAttribute(tgtColName), fakeSourcePlan, s"$typ clause")
+                resolveOrFail(functions.expr(s"`$tgtColName`").expr, fakeSourcePlan, s"$typ clause")
```

Comment:
hey @parisni while merging this, I made a tiny change. I replaced this with `UnresolvedAttribute.quotedString(s"`$tgtColName`")` to keep it consistent with other code in this file which uses only spark-catalyst (`functions` stuff is spark-sql).

Reply:
@tdas yeah I wasn´t very clear with the unresolvedAttribute stuff. Thanks